src/cpp/src/vision_encoder.cpp

```diff
@@ -0,0 +1,217 @@
+    std::vector<std::vector<clip_image_f32>> preprocessed{imgs.size()};
+    std::transform(imgs.begin(), imgs.end(), preprocessed.begin(), [&ctx_clip](const std::vector<clip_image_u8>& row) {
+        std::vector<clip_image_f32> processed_row{row.size()};
+        std::transform(row.begin(), row.end(), processed_row.begin(), [&ctx_clip](const clip_image_u8& raw) {
+            return clip_image_preprocess(ctx_clip, raw);
+        });
+        return processed_row;
+    });
+
+    const clip_image_f32& resized_preprocessed = preprocessed.at(0).at(0);
+    ov::Tensor input_tensor{ov::element::f32, {1, 3, size_t(resized_preprocessed.ny), size_t(resized_preprocessed.nx)}, (void*)(resized_preprocessed.buf.data())};
+    encoder.set_input_tensor(input_tensor);
+    encoder.infer();
+    ov::Tensor output_tensor = encoder.get_output_tensor();
+    ov::Tensor resized_source{output_tensor.get_element_type(), output_tensor.get_shape()};
+    output_tensor.copy_to(resized_source);
+    HeightWidth resized_source_size{resized_preprocessed.ny / patch_size, resized_preprocessed.nx / patch_size};
+
+    HeightWidth size{
+        size_t(preprocessed.at(1).at(0).ny),
+        size_t(preprocessed.at(1).at(0).nx)
+    };
+    ov::Tensor batched{ov::element::f32, {(preprocessed.size() - 1) * preprocessed.at(1).size(), 3, size.height, size.width}};
+    float* batched_data = batched.data<float>();
+    size_t batch_offset = 0;
+    size_t values_in_elem = 3 * size.height * size.width;
+    std::vector<HeightWidth> sliced_sizes;
+    for (size_t row = 1; row < preprocessed.size(); ++row) {
+        for (const clip_image_f32& elem : preprocessed.at(row)) {
+            std::copy_n(elem.buf.begin(), values_in_elem, batched_data + batch_offset);
+            sliced_sizes.push_back({elem.ny / patch_size, elem.nx / patch_size});
+            batch_offset += values_in_elem;
+        }
+    }
+    encoder.set_input_tensor(batched);
+    encoder.infer();
+    const ov::Tensor& encoded = encoder.get_output_tensor();
+    const ov::Shape& plain = encoded.get_shape();
+    struct SharedTensorAllocator {
+        const ov::Tensor tensor;
+        void* allocate(size_t bytes, size_t) {return bytes <= tensor.get_byte_size() ? tensor.data() : nullptr;}
+        void deallocate(void*, size_t, size_t) {}
+        bool is_equal(const SharedTensorAllocator& other) const noexcept {return this == &other;}
+    };
+    ov::Tensor reshaped{encoded.get_element_type(), {preprocessed.size() - 1, preprocessed.at(1).size(), plain.at(1), plain.at(2)}, SharedTensorAllocator{encoded}};
+    return {resized_source, resized_source_size, reshaped, sliced_sizes};
+}
+}
+
+VisionEncoder::VisionEncoder(const std::filesystem::path& model_dir, const std::string& device, const ov::AnyMap device_config, ov::Core core) :
+    VisionEncoder{
+        core.compile_model(
+            // CPU only because of 146022.
+            model_dir / "openvino_vision.xml", "CPU", device_config
```

Comment:
I have checked with GPU, it works! 
```suggestion
            model_dir / "openvino_vision.xml", device, device_config
```

Reply:
What's that another place?
Why would CPU inference, wherever it is, break GPU?